plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -299,11 +312,10 @@ public ConnectorPageSource createPageSource(
         List<IcebergColumnHandle> readColumns = dataPageSource.getReaderColumns()
                 .map(readerColumns -> readerColumns.get().stream().map(IcebergColumnHandle.class::cast).collect(toList()))
                 .orElse(requiredColumns);
-        DeleteFilter<TrinoRow> deleteFilter = new TrinoDeleteFilter(
-                dummyFileScanTask,
-                tableSchema,
-                readColumns,
-                fileIO);
+
+        Optional<RowPredicate> deletePredicate = deleteFilters.stream()
+                .map(filter -> filter.createPredicate(readColumns))
+                .reduce(RowPredicate::and);
```

Comment:
This creates intermediate nodes. It would be more efficient to have a composite flat "and" predicate with a list of sub-predicates

Reply:
I considered that originally, but I'm not sure it would be more efficient. We expect to only have one or a few filters, so with inlining, the JVM might be able to de-virtualize and flatten the calls into a single expression. The flat list seems to have more overhead for the common cases.

Reply:
We can optimize this later if it becomes an issue. An obvious one is to combine all the deletes into a single bitmap. For equality, we could use method handle combinators to combine the predicates, and possibly also to "compile" the equality comparison expression.

Reply:
I combined position deletes into a single bitmap.